### ISSUE TITLE
vagrant-winrm check needs to be modified

### ISSUE THREAD

**Issue by cheeseplus** [test-kitchen/kitchen-vagrant#372]
## Description

opened on behalf of @safematix

Vagrant has detected the `vagrant-winrm` plugin. Vagrant ships with
WinRM support builtin and no longer requires the `vagrant-winrm` plugin.

kitchen Version: 1.23.2
Chef Development Kit Version: 3.2.30
Vagrant Version: 2.1.4

## Platform Version

Ubuntu 18.04

## Replication Case

Install vagrant 2.1.4 and vagrant-winrm plugin.

## Kitchen Output

Kitchen/Vagrant output with vagrant-winrm plugin:

```ruby
STDERR: WARNING: Vagrant has detected the `vagrant-winrm` plugin. Vagrant ships with
WinRM support builtin and no longer requires the `vagrant-winrm` plugin. To
prevent unexpected errors please uninstall the `vagrant-winrm` plugin using
the command shown below:

  vagrant plugin uninstall vagrant-winrm

To disable this warning, set the environment variable `VAGRANT_IGNORE_WINRM_PLUGIN`
```

Kitchen output without vagrant-winrm plugin:

```ruby
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::UserError
>>>>>> Message: WinRM Transport requires the vagrant-winrm Vagrant plugin to properly communicate with this Vagrant VM. Please install this plugin with: `vagrant plugin install vagrant-winrm' and try again.
>>>>>> ----------------------
```

The latter warning is coming from kitchen-vagrant and we can certainly adjust the check but this is the wrong repo for the issue/work.

https://github.com/test-kitchen/kitchen-vagrant/blob/d519b679abe8963bb68c897fd16370d2aeef95ca/lib/kitchen/driver/vagrant.rb#L547-L565

The check in upstream Vagrant was only very recently added in 2.1.3.

**Comment by cheeseplus** [comment]
kitchen-vagrant relies on the output of `vagrant winrm-config` to do it's business so getting rid of the check for newer versions of vagrant just moves the error.

```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [Expected process to exit with [0], but received '1'
---- Begin output of vagrant winrm-config ----
STDOUT: Usage: vagrant [options] <command> [<args>]

    -v, --version                    Print the version and exit.
    -h, --help                       Print this help.

Common commands:
     box             manages boxes: installation, removal, etc.
     connect         connect to a remotely shared Vagrant environment
     destroy         stops and deletes all traces of the vagrant machine
     global-status   outputs status Vagrant environments for this user
     halt            stops the vagrant machine
     help            shows the help for a subcommand
     init            initializes a new Vagrant environment by creating a Vagrantfile
     login           log in to HashiCorp's Vagrant Cloud
     package         packages a running vagrant environment into a box
     plugin          manages plugins: install, uninstall, update, etc.
     port            displays information about guest port mappings
     powershell      connects to machine via powershell remoting
     provision       provisions the vagrant machine
     push            deploys code in this environment to a configured destination
     rdp             connects to machine via RDP
     reload          restarts vagrant machine, loads new Vagrantfile configuration
     resume          resume a suspended vagrant machine
     share           share your Vagrant environment with anyone in the world
     snapshot        manages snapshots: saving, restoring, etc.
     ssh             connects to machine via SSH
     ssh-config      outputs OpenSSH valid configuration to connect to the machine
     status          outputs status of the vagrant machine
     suspend         suspends the machine
     up              starts and provisions the vagrant environment
     validate        validates the Vagrantfile
     version         prints current and latest Vagrant version

For help on any individual command run `vagrant COMMAND -h`

Additional subcommands are available, but are either more advanced
or not commonly used. To see all subcommands, run the command
`vagrant list-commands`.
STDERR:
---- End output of vagrant winrm-config ----
Ran vagrant winrm-config returned 1] on default-windows-10
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

**Comment by dayglojesus** [comment]
I think this can be closed now as of #379.
